internal/cmd/controller/reconciler/bundle_status.go

```diff
@@ -41,64 +32,26 @@ func updateDisplay(status *fleet.BundleStatus) {
 	status.Display.State = string(summary.GetSummaryState(status.Summary))
 }
 
-// setResourceKey updates status.ResourceKey from the bundle, by running helm template (does not mutate bundle)
-func setResourceKey(ctx context.Context, status *fleet.BundleStatus, bundle *fleet.Bundle, manifest *manifest.Manifest, isNSed func(schema.GroupVersionKind) bool) error {
-	seen := map[fleet.ResourceKey]struct{}{}
-
-	// iterate over the defined targets, from "targets.yaml", not the
-	// actually matched targets to avoid duplicates
-	for i := range bundle.Spec.Targets {
-		opts := options.Merge(bundle.Spec.BundleDeploymentOptions, bundle.Spec.Targets[i].BundleDeploymentOptions)
-		objs, err := helmdeployer.Template(ctx, bundle.Name, manifest, opts)
-		if err != nil {
-			logrus.Infof("While calculating status.ResourceKey, error running helm template for bundle %s with target options from %s: %v", bundle.Name, bundle.Spec.Targets[i].Name, err)
+// setResourceKey updates status.ResourceKey from the bundleDeployments
+func setResourceKey(status *fleet.BundleStatus, allTargets []*target.Target) {
+	keys := []fleet.ResourceKey{}
+	for _, target := range allTargets {
+		if target.Deployment == nil {
 			continue
 		}
-
-		for _, obj := range objs {
-			m, err := meta.Accessor(obj)
-			if err != nil {
-				return err
-			}
+		if target.Deployment.Namespace == "" {
+			logrus.Infof("Skipping bundledeployment with empty namespace %v", target.Deployment)
```

Comment:
I find this log message a bit confusing, as if `%v` were a placeholder for an empty namespace...
Also, what information is really relevant for logging here? Does a full bundle deployment body make sense, or should we instead log its name?
How about this? Happy to discuss.
```suggestion
			logrus.Infof("Skipping bundledeployment %q with empty namespace", target.Deployment.Name)
```

Reply:
Agree. This is just to log when bundle deployment is skipped. I've applied the suggested changes